haskell/cabal.bzl

```diff
@@ -166,11 +177,14 @@ def _haskell_cabal_library_impl(ctx):
         "_install/lib/libHS{}-ghc{}.{}".format(name, hs.toolchain.version, _so_extension(hs)),
         sibling = cabal,
     )
+    (tool_inputs, tool_input_manifests) = ctx.resolve_tools(tools = ctx.attr.tools)
```

Comment:
Fascinating, I didn't know about [`resolve_tools`](https://docs.bazel.build/versions/master/skylark/lib/ctx.html#resolve_tools). Looks like [it was introduced in 0.23.0](https://github.com/bazelbuild/bazel/commit/52446a12b7435e850347dda144566280625cf498), there's a [design doc](https://docs.google.com/document/d/1xPsvTY-vWav9zX--7ieXjUilcl7M46m-88_oNV0QhEU/edit#) for anyone interested.